src/openrct2/platform/Android.cpp

```diff
@@ -62,4 +62,51 @@ bool platform_get_steam_path(utf8* outPath, size_t outSize)
     return false;
 }
 
+AndroidClassLoader::AndroidClassLoader()
+{
+    log_info("Obtaining JNI class loader");
+
+    // This is a workaround to be able to call JNI's ClassLoader from non-main
+    // thread, based on https://stackoverflow.com/a/16302771
+    // Apparently it's OK to use it from across different thread, but JNI
+    // only looks for ClassLoader in the _current_ thread and fails to find
+    // it when searched for from a native library's non-main thread.
+
+    // If we're here, SDL's JNI_OnLoad has already been called and set env
+    JNIEnv* env = (JNIEnv*)SDL_AndroidGetJNIEnv();
+
+    // First load some our random class
```

Comment:
This looks ungrammatical to me.

Reply:
Same, but can you provide a better alternative?

Reply:
I would change it to `// While we use 'MainActivity' here, this workaround will work with any class.`, then.